datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -217,19 +248,13 @@ impl LogicalPlanBuilder {
 
     /// Convert a table provider into a builder with a TableScan
     pub fn scan_with_filters(
-        table_name: impl Into<String>,
+        table_name: impl Into<OwnedTableReference>,
         table_source: Arc<dyn TableSource>,
         projection: Option<Vec<usize>>,
         filters: Vec<Expr>,
     ) -> Result<Self> {
         let table_name = table_name.into();
 
-        if table_name.is_empty() {
-            return Err(DataFusionError::Plan(
-                "table_name cannot be empty".to_string(),
-            ));
-        }
```

Comment:
does this check need to be added to `TableReference` related methods? didn't consider this before

Reply:
🤔  the `TableReference` methods are infallable (don't return Result) so making them fallable would be a substantial change. I will restore this check and add some test coverage. Good spot.

Reply:
in c2344898f